packages/invoices/src/OutboundShipment/DetailView/DetailView.tsx

```diff
@@ -66,6 +74,23 @@ export const DetailView: FC = () => {
     [sortBy]
   );
 
+  const onChangeSelectedItem = (newItem: Item | null) => {
+    if (!newItem) return setSelectedItem(newItem);
+
+    // Try and find the outbound summary row that matches the new item
+    const item = draft.items.find(
+      summaryItem => summaryItem.itemId === newItem.id
+    );
+
+    // If we found it, set the selected item.
+    if (item) {
+      setSelectedItem(item);
+    } else {
+      // otherwise, set the selected item to a newly created summary row.
+      setSelectedItem(itemToSummaryItem(newItem));
+    }
+  };
```

Comment:
this is the callback for the modal. It will be simplified when we get to filtering out existing items in the invoice out of the items list